src/librustc_mir/interpret/traits.rs

```diff
@@ -140,7 +140,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // to determine the type.
         let drop_instance = self.memory.get_fn(drop_fn)?.as_instance()?;
         trace!("Found drop fn: {:?}", drop_instance);
-        let fn_sig = drop_instance.ty(*self.tcx).fn_sig(*self.tcx);
+        let fn_sig = drop_instance.monomorphic_ty(*self.tcx).fn_sig(*self.tcx);
```

Comment:
This is part of the interpreter, why is it not using the param env?

Reply:
Fixed